router/pkg/mcpserver/schema_compiler.go

```diff
@@ -0,0 +1,78 @@
+package mcpserver
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/santhosh-tekuri/jsonschema/v6"
+	"github.com/wundergraph/cosmo/router/pkg/schemaloader"
+	"go.uber.org/zap"
+)
+
+// SchemaCompiler handles JSON schema compilation for operations
+type SchemaCompiler struct {
+	logger *zap.Logger
+}
+
+// NewSchemaCompiler creates a new schema compiler with the given logger
+func NewSchemaCompiler(logger *zap.Logger) *SchemaCompiler {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+	return &SchemaCompiler{
+		logger: logger,
+	}
+}
+
+// CompileSchema compiles a JSON schema for a single operation
+func (sc *SchemaCompiler) CompileSchema(op schemaloader.Operation) (*jsonschema.Schema, error) {
+	if len(op.JSONSchema) == 0 {
+		return nil, nil
+	}
+
+	c := jsonschema.NewCompiler()
+
+	// Load the JSON schema from the operation
+	schema, err := jsonschema.UnmarshalJSON(io.NopCloser(bytes.NewReader(op.JSONSchema)))
```

Comment:
I don't think you need `io.NopCloser` here.

Reply:
Good catch